ui/ui-components/components/property/PropertySampleRecord.tsx

```diff
@@ -0,0 +1,87 @@
+import { useState, useCallback, useMemo } from "react";
+import { useDebouncedValue } from "../../hooks/useDebouncedValue";
+import { Actions, Models } from "../../utils/apiData";
+import SampleRecordCard, {
+  SampleRecordCardProps,
+} from "../record/SampleRecordCard";
+
+interface Props {
+  property: Models.PropertyType;
+  localFilters: Actions.PropertyView["property"]["filters"];
+}
+
+const PropertySampleRecord: React.FC<
+  Props & Omit<SampleRecordCardProps, "fetchRecord" | "highlightProperty">
+> = ({ property, localFilters, ...props }) => {
+  const { execApi } = props;
+  const debouncedProperty = useDebouncedValue(property, 1000);
+
+  const [errorMessage, setErrorMessage] = useState("");
+
+  const reloadKey = useMemo(() => {
+    return JSON.stringify([
+      debouncedProperty.id,
+      debouncedProperty.type,
+      debouncedProperty.unique,
+      debouncedProperty.isArray,
+      debouncedProperty.options,
+      debouncedProperty.filters,
+      JSON.stringify(localFilters),
+    ]);
+  }, [
+    debouncedProperty.id,
+    debouncedProperty.type,
+    debouncedProperty.unique,
+    debouncedProperty.isArray,
+    JSON.stringify(debouncedProperty.options),
+    JSON.stringify(debouncedProperty.filters),
+    JSON.stringify(localFilters),
+  ]);
```

Comment:
Couple of options to improve this:

a. Use https://www.npmjs.com/package/fast-json-stringify over `JSON.stringify`
b. Introduce https://www.npmjs.com/package/use-deep-compare

Thoughts?

Reply:
Yeah... the need to stringify these objects to notice changes took a long time to debug, and this solution certainly isn't great.  

As an alternative approach, what if we had a new `lastChangeAt` timestamp useState variable on the parent form.  We could bind to any change on the form (`form.addEventListener('change',() => setLastChangeAd(new Date())`) and then we could just use this timestamp to know it was time to reload

Reply:
Or... I guess we could just stringify the whole thing -> `JSON.stringify(debouncedProperty)`.  That would probably also get the job done